lockfile_generator/src/lib.rs

```diff
@@ -34,7 +33,12 @@ pub trait Generator {
     ///
     /// This will ignore all existing lockfiles and create a new lockfile based
     /// on the current project configuration.
-    fn generate_lockfile(&self, project_path: &Path) -> Result<String> {
+    fn generate_lockfile(&self, manifest_path: &Path) -> Result<String> {
+        let canonicalized = fs::canonicalize(manifest_path)?;
+        let project_path = canonicalized
+            .parent()
+            .ok_or_else(|| Error::InvalidManifest(manifest_path.to_path_buf()))?;
```

Comment:
This API change makes the assumption that there will be a manifest in the root of the project for every single ecosystem.

It's probably going to be fine, but I did want to bring it up at least.

Reply:
It's a fair point. This code was just moved over from `commands/parse.rs`, so I've basically just moved the assumption (across an API barrier, which is significant).

But we can always change this if we introduce an ecosystem that breaks the assumption